carbonmark-api/src/utils/ICR/icr.utils.ts

```diff
@@ -0,0 +1,50 @@
+import fetch from "node-fetch";
+import type { NetworkParam } from "../../../src/models/NetworkParam.model";
+import type { IcrProject } from "../ICR/icr.types";
+import { ICR_API } from "./ICR_API_endpoints";
+
+export const convertIcrCountryCodeToName = (code: string) => {
+  if (!code) return;
+  const regionNames = new Intl.DisplayNames(["en"], { type: "region" });
+  const country = regionNames.of(code);
+  return country;
+};
+
+export const fetchAllICRProjects = async (
+  network: NetworkParam
+): Promise<IcrProject[]> => {
+  const { ICR_API_URL, ICR_API_KEY } = ICR_API(network);
+  let allProjects: IcrProject[] = [];
+  let page = 0;
+  const limit = 50;
+  let shouldFetchProjects = true;
+
+  while (shouldFetchProjects) {
+    const response = await fetch(
```

Comment:
Are we doing that for each request to the `/projects` endoint ? This looks like a real performance bottleneck

Reply:
It is; the ICR projects endpoint has a 50 endpoint limit. This entire issue will be removed though now that the ICR projects are in the cms.

Reply:
refactored to use project cms data and removed